### ISSUE TITLE
Issues relating to Prompts Redescription

### ISSUE THREAD

**Issue by Arlo0321** [MirrorDiffusion/MirrorDiffusion#1]
Dear Authors, Would you please point me to the location of the corresponding code for Eq. 3 and Eq. 4 in the paper?

![image](https://github.com/MirrorDiffusion/MirrorDiffusion/assets/51200306/51e6c3f3-e3fd-4c30-91bf-5f4c732cb029)

I tried to find it, but it didn't feel right. In lines 115 to 119 of src/utils/pipe_null.py, it seems that uncond_embeddings is optimized here, not text prompt embedding as indicated in the paper. In addition to this, I have not found where λ corresponds to Eq. 4.

![image](https://github.com/MirrorDiffusion/MirrorDiffusion/assets/51200306/f72d0810-b97b-47a6-a6ff-81f2fef325b4)

**Comment by YupeiLin2388** [comment]
I'm sorry for the confusion caused by the irregularity of my code.

During the reconstruction, our method does not use classifier-free guidance, so both `uncond_embeddings` and `cond_embeddings` represent text prompt embeddings here. The following code demonstrates that the reconstruction stage and editing stage are based on the `uncond_embeddings` variable：

```python
with torch.no_grad():
    prompt_embeds[0] = uncond_embeddings.clone().detach()
    prompt_embeds_edit[0] = uncond_embeddings.clone().detach()
    prompt_embeds[1] = uncond_embeddings.clone().detach()
    prompt_embeds_edit[1] = uncond_embeddings.clone().detach()
    prompt_embeds_edit[1:2] +=0.65*edit_dir
```

For the parameter $\lambda$ in Equation 4, we aim to represent the magnitude of the MSE Loss (`inv_loss`). In this paper, $\lambda$ is set to 1, so we have omitted the parameter $\lambda$. If you want to change the parameter $\lambda$, you can do so by adding a parameter in `inv_loss = nn.functional.mse_loss(latents_prev_rec, latent_prev)` or by modifying the `lr` parameter in the `optimizer`.
